pandas/indexes/base.py

```diff
@@ -86,6 +86,12 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    # required for backward compat, because PI can't be instantiated with
+    # ordinals through __new__ GH #13277
+    if issubclass(cls, ABCPeriodIndex):
+        if d['data'].dtype == 'int64':
+            values = d.pop('data')
```

Comment:
Why do we use `__reduce__` for Indexes rather than `__getstate__`, like we do in the rest of the library? 
I think the pickling code for Indexes could be _much_ simpler. More than happy to do this in a follow up PR. I don't want to get into the habit of blindly making changes I don't see the logic in though